doc/introduction/data.rst

```diff
@@ -0,0 +1,162 @@
+.. role:: html(raw)
+   :format: html
+
+.. _intro_ref_data:
+
+Quantum Datasets
+================
+
+PennyLane provides the :mod:`~.data` subpackage to download, create, store and manipulate quantum datasets within the PennyLane framework.
+
+Loading Datasets in PennyLane
+-----------------------------
+
+We can access data of a desired type with the :func:`~pennylane.data.load` function. This downloads the desired
+dataset or loads it from local storage if previously downloaded.
+
+To specify the dataset to be loaded, the data category (``data_name``) must be
+specified, alongside category-specific keyword arguments. For the full list
+of available datasets, please see the dataset library documentation.
+The :func:`~pennylane.data.load` returns a list with the desired data which can then be used within
+the usual PennyLane workflow.
+
+.. code-block:: python
+
+    >>> H2_dataset = qml.data.load(data_name="qchem", molname="H2", basis="6-31G", bondlength="0.46")
+    >>> print(H2_dataset)
+    [<pennylane.data.dataset.Dataset object at 0x7f14e4369640>]
+
+When we only want to download portions of a large dataset, we can specify the desired properties  (referred to as _attributes_).
+For example, we can download only the molecule and hamiltonian of a dataset as follows:
+
+.. code-block:: python
+
+    >>> H2_hamiltonian = qml.data.load(data_name='qchem',molname='H2', basis='6-31G', bondlength='0.46', attributes=['molecule','hamiltonian'])
```

Comment:
We now support bond length as `float`, might want to reflect it here.

Reply:
Also, this would be a list, right? Why aren't we accessing the 0th index and then the Hamiltonian? Or are there any changes to reading specific attributes?

Reply:
we explained the list above, I'd just add `[0]` to the end of this line personally. plus, our users might get into that habit if they are loading one dataset at a time, might as well show the type of stuff we've been doing internally.

Reply:
No changes, I added the `[0]`